### PR TITLE
Added UserRolesController

### LINKED ISSUE #44: [DEV] UserRole Controller

### PR DESCRIPTION
**Pull request by L3viathon**
Unsure if mapping to endpoints is correct...

### CODE REVIEW

**Comment by L3viathon** [comment]
closes #44

**Review by FullSol — APPROVED** [review]
Looks great!

File: src/main/java/com/amplifier/controllers/UserRolesController.java [hunk 1]
```diff
@@ -1,5 +1,68 @@
...
+@RestController
+@RequestMapping("/api/v1")
+@Api (value = "UserRolesRestController", description = "REST controller related to UserRole Entities")
 public class UserRolesController {
 
+    @Autowired
+    private UserRolesService userRolesService;
+
+    @GetMapping("/roles")
+    @ApiOperation(value = "Find all user roles.")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/UserRolesController.java]: Please provide notes for the API's documentation.

Comment (FullSol) [inline src/main/java/com/amplifier/controllers/UserRolesController.java]: - Please provide the response body for the API documentation.

File: src/main/java/com/amplifier/controllers/UserRolesController.java [hunk 2]
```diff
@@ -1,5 +1,68 @@
...
+
+    @ApiOperation(value = "Find user role by id number", notes = "Provide an id to lookup a specific user role from the API", response = UserRole.class)
+    @GetMapping("/role?id={id}")
+    public @ResponseBody UserRole getById(@RequestParam(value = "id")int id) {
+        System.out.println("TEST: " + userRolesService.getUserRoleById(id));
+        return userRolesService.getUserRoleById(id);
+    }
+
+    @PostMapping("/role")
+    @ApiOperation(value = "Create new user role entity")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/UserRolesController.java]: API Operation note

File: src/main/java/com/amplifier/controllers/UserRolesController.java [hunk 3]
```diff
@@ -1,5 +1,68 @@
...
+    public @ResponseBody ClientMessage createUserRole(@RequestBody UserRole userRole) {
+        return userRolesService.addUserRole(userRole) ? CREATION_SUCCESSFUL : CREATION_FAILED;
+    }
+
+    @PutMapping("/role")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/UserRolesController.java]: Change this to patch. Put will create a new one OR update.

File: src/main/java/com/amplifier/controllers/UserRolesController.java [hunk 4]
```diff
@@ -1,5 +1,68 @@
...
+    @ApiOperation(value = "Update user role entity by id.")
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/UserRolesController.java]: API Operation notes

File: src/main/java/com/amplifier/controllers/UserRolesController.java [hunk 5]
```diff
@@ -1,5 +1,68 @@
...
+    @ApiOperation(value = "Find all user roles.")
+    public @ResponseBody List<UserRole> getAll(){
+        return userRolesService.getAllUserRoles();
+    }
+
+    @ApiOperation(value = "Find user role by id number", notes = "Provide an id to lookup a specific user role from the API", response = UserRole.class)
+    @GetMapping("/role?id={id}")
+    public @ResponseBody UserRole getById(@RequestParam(value = "id")int id) {
+        System.out.println("TEST: " + userRolesService.getUserRoleById(id));
+        return userRolesService.getUserRoleById(id);
```
Comment (FullSol) [inline src/main/java/com/amplifier/controllers/UserRolesController.java]: Please change the naming convention.

**Review by FullSol — CHANGES_REQUESTED** [review]
Please review and make changes per the noted comments.

Thanks!

**Review by pyaeger — APPROVED** [review]
Reviewed together. Calvin requested changes now pending. Thanks!

**Review by FullSol — APPROVED** [review]

**Pull request merged**